### PR TITLE
feat(avalonia): Map Tile Animation Type 1 image preview + Import/Export (Ref #1602)

### PR DESCRIPTION
**Pull request by laqieer**
## Summary
The Avalonia **Map Tile Animation Type 1** editor previously edited only scalar fields (it had **no image surface at all**), while WinForms `MapTileAnimation1Form` and the sibling Avalonia anime2 view have image preview + Import/Export. This PR closes that asymmetry: a live **GbaImageControl preview**, single-PNG **Export/Import**, and `.mapanime1.txt` batch **Export All / Import All**.

New Core seam `MapTileAnimation1ImageCore` honours the **inverse anime1 schema** (per CLAUDE.md): the graphics block is **RAW 4bpp at the entry's `+4` pointer**, sized by the **`+2` u16 length**, **NOT LZ77** (unlike anime2's `+0`). It reuses the existing CLI `--kind=mapanime1gfx` RAW-4bpp model and `ImageImportCore.WriteRawToROM` (never `WriteCompressedToROM`).

**`+2` length authority** (the WF semantics, confirmed in the Copilot plan review): single import keeps the entry's existing `+2` length authoritative — it rejects a PNG unless `height == CalcEntryHeight(currentLength)`, writes only the raw bytes to `+4`, and **leaves `+2` unchanged**. Batch import honours the explicit 3rd "length" column (truncate) when present, else the full encoded length, every write `<= 0xFFFF` guarded.

All ROM mutations: validate-before-mutate + ambient undo + defensive snapshot **byte-identical fault restore** (#885/#923).

## Scope
`Ref #1602 (partial — composited-map GIF follow-up)` — delivers the full **data I/O surface**: preview + single PNG Export/Import + `.mapanime1.txt` batch (lossless per-entry PNG round-trip). The WinForms "Export All → .gif" variant composites each frame onto the rendered chapter map via WinForms `MapSettingForm.DrawMap` (not in Core); that composited-map GIF is **deferred** as a follow-up. The `.mapanime1.txt` batch is the lossless data path and fully closes the editing gap.

## Screenshots
Map Tile Animation 1 editor

<sub>Hosted on the repo's permanent **issue-assets** asset branch (a stable, non-feature branch dedicated to PR/issue screenshots — it persists after this feature branch is deleted, so the link does not break). The same PNG is also committed in this PR at `pr-screenshots/pr1602-maptileanim1-fe8u.png`, so it additionally resolves at the `master/pr-screenshots/...` URL once this PR merges (direct push to master is rule-protected, so it cannot pre-exist on master before merge).</sub>

The new **Tile Animation Image** panel + **Sample Palette** combo on the Avalonia Map Tile Animation Type 1 editor (FE8U, ANIME1 "The Fall of Renais", entry 0x00 populated). Captured via headless `--screenshot-all`. (The Export/Import/Export All/Import All buttons render directly below the preview, within the same panel — see `MapTileAnimation1View.axaml`.)

## GUI Test Report
| Area | Test | Result |
|------|------|--------|
| Editor launch | `--screenshot-all` opens MapTileAnimation1View on FE8U without crash | PASS |
| New panel | "Tile Animation Image" panel + Sample Palette combo render with populated entry data | PASS |
| Preview control | `GbaImageControl PreviewImage` present + AutomationId `MapTileAnimation1_Preview_Image` | PASS (headless) |
| Buttons present | Export / Import / Export All / Import All present with AutomationIds | PASS (headless) |
| Sample Palette combo | 16 sub-palette entries, default index 0 | PASS (headless) |

## Test plan
- [x] Core: `CalcEntryHeight` matches the WF `CalcHeight(256,len)` formula (incl. round-up + 8-align edge cases)
- [x] Core: single `ImportEntryImage` keeps `+2` length **unchanged** and repoints `+4` to fresh raw bytes
- [x] Core: imported RAW bytes round-trip **byte-identical** to the independently-encoded tiles at `+4`
- [x] Core: wrong **height** image → rejected, ROM **byte-identical** (no mutation)
- [x] Core: wrong **width** image → rejected, ROM **byte-identical** (no mutation)
- [x] Core: free-space exhaustion fault → ROM restored **byte-identical**
- [x] Core: `RenderEntryImage` returns a 256×H image for a valid entry, null for a bad/zero pointer (never throws)
- [x] Avalonia headless: `MapTileAnimation1View` instantiates; `PreviewImage` GbaImageControl present
- [x] Avalonia headless: Sample-palette combo populated (16 entries, index 0)
- [x] Avalonia headless: all 4 image buttons + preview + combo carry their AutomationIds
- [x] 13 new Core.Tests + 13 new Avalonia headless tests all green locally
- [x] GUI proof: headless screenshot of the affected editor with populated data (attached)

**Test suites** (worktree env: `config/patch2` submodule + bundled ROMs not checked out — CI is authoritative):
- Core.Tests: new 13/13 PASS; 88 pre-existing env failures (patch2-dependent Skill/Struct/ExportFilter + ROM tests — unrelated to this change, no new file touched)
- Avalonia.Tests: new 13/13 PASS; 5 pre-existing env failures (ROM-path + Skill-system VMs — unrelated)
- FEBuilderGBA.Tests (net9.0-windows): builds clean; no WinForms files touched

## Plan
Plan + Copilot plan-review (length-authority blocking finding resolved in rev 2): https://github.com/laqieer/FEBuilderGBA/issues/1602#issuecomment-4816160052

Generated with Claude Code (Opus 4.8 · claude-opus-4-8[1m])

### CODE REVIEW

**Review by laqieer — COMMENTED** [review]
Review findings:

1. **Blocking: the required GUI screenshot does not render.** The PR body uses `https://raw.githubusercontent.com/laqieer/FEBuilderGBA/master/pr-screenshots/pr1602-maptileanim1-fe8u.png`, but that URL currently returns 404, so the PR description does not contain a rendered image proving the feat GUI change works. Because this PR starts with `feat` and modifies `FEBuilderGBA.Avalonia/`, this fails the screenshot gate. Please either use a GitHub-uploaded asset or a default-branch raw/blob URL that already exists and renders.

2. **Blocking: batch import can corrupt entries by reusing the same free-space run.** In `FEBuilderGBA.Core/MapTileAnimation1ImageCore.cs:321`, `ImportBatchTxt` writes each frame with `ImageImportCore.FindAndWriteData`, then writes the rebuilt table with `WriteRawToROM` at line 336. `ROM.FindFreeSpace` treats both `0x00` and `0xFF` runs as free (`FEBuilderGBA.Core/Rom.cs:1055-1062`), so a valid all-transparent frame encoded as all `0x00` remains discoverable as free space. A later frame write or the final table write can allocate the same address, overwrite the earlier image, and leave the manifest table pointing multiple entries at corrupted/shared data. This also means the batch Import All path has no regression coverage for the user-visible feature and does not meet the plan's recycle/reservation expectation. Please reserve/recycle written regions like the WinForms `RecycleAddress` path (or otherwise guarantee monotonic/non-overlapping allocations) and add a batch test with an all-zero frame followed by another frame/table write.

The GUI Test Report section exists with a results table, and the Test plan checklist items are all checked.

Copilot CLI: 1.0.66-1
Model: GPT-5.5 (gpt-5.5)

File: FEBuilderGBA.Core/MapTileAnimation1ImageCore.cs [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+                {
+                    string line = lines[i];
+                    if (U.IsComment(line) || U.OtherLangLine(line)) continue;
+                    line = U.ClipComment(line);
+                    if (line == "") continue;
+                    string[] sp = line.Split('\t');
+                    if (sp.Length < 2) continue;
+
+                    uint wait = U.atoi(sp[0]);
+                    string imgPath = Path.Combine(dir, sp[1].Trim());
```
[the remark this answers is not in the document]
Comment (laqieer) [reply FEBuilderGBA.Core/MapTileAnimation1ImageCore.cs]: Fixed in deb57efb4: the manifest `wait` column is now range-checked and **rejected** (not silently truncated) when > 0xFFFF, with no ROM mutation. Covered by `ImportBatchTxt_WaitOverU16_Rejected_NoMutation`.

File: FEBuilderGBA.Core/MapTileAnimation1ImageCore.cs [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+
+            string[] lines;
+            try { lines = File.ReadAllLines(txtPath); }
+            catch (Exception ex) { return R._("Cannot read manifest: {0}", ex.Message); }
+
+            string dir = Path.GetDirectoryName(txtPath) ?? ".";
+            byte[] palette = ImageUtilCore.GetPalette(rom, palOffset + (uint)(paletteIndex * 16 * 2), 16);
+
+            byte[] snap = (byte[])rom.Data.Clone();
+            try
```
[the remark this answers is not in the document]
Comment (laqieer) [reply FEBuilderGBA.Core/MapTileAnimation1ImageCore.cs]: Fixed in deb57efb4: the ~32MB `rom.Data.Clone()` snapshot is now taken **only just before the first (and only) ROM write**, after all blocks are validated/encoded — so the validation-only no-op path (missing PNG, bad dims, wait/length overflow) pays no full clone. The batch also now writes all blocks + the table as ONE contiguous allocation, so the single snapshot is a self-consistent restore point.

**Review by laqieer — COMMENTED** [review]
Re-review complete: the two prior blocking concerns are resolved.

- The screenshot now uses the permanent `issue-assets` raw URL, and I verified it returns a valid PNG (`200 OK`, `Content-Type: image/png`, PNG signature, 35,589 bytes). The PNG is also included in the PR file list at `pr-screenshots/pr1602-maptileanim1-fe8u.png`, so it will land on `master` at merge.
- `ImportBatchTxt` now validates/encodes all frames before mutation, delays the ROM snapshot until the first write, writes all image blocks plus the rebuilt entry table through one contiguous allocation, and then fixes up the table pointers. This addresses the all-0x00 frame alias/overwrite risk from the previous review.
- The prior inline concerns are addressed: `wait > 0xFFFF` is rejected with no mutation, and the validation-only failure path no longer clones the full ROM snapshot.
- Focused checks passed locally: `MapTileAnimation1ImageCoreTests` (16/16) and `MapTileAnimation1ImageViewHeadlessTests` (13/13).

No blocking findings remain from my review.

Copilot CLI: 1.0.66-1
Model: GPT-5.5 (gpt-5.5)
